### ISSUE TITLE
Discord module disconnecting and not properly reconnecting

### ISSUE THREAD

**Issue by Teeko1993** [Budabot/Tyrbot#27]
Occasionally, the discord module will error out with: 
INFO - websockets.protocol - Failing the WebSocket connection: 1011 
INFO - discord.gateway - Websocket closed with 1006 (), attempting a reconnect.
INFO - discord.client - Got ResumeWebSocket

The bot then doesn't automatically reconnect to discord, but the bot thinks it is connected. So the only way to then connect to discord properly is to restart the bot.

**Comment by bigwheels16** [comment]
I added `!discord disconnect`.  As a quick fix until a better fix is found, hopefully you can `!discord disconnect` and then `!discord connect` to fix it without restarting.

**Comment by Dustify** [comment]
Had similar issues with the discord API in .NET, will either wait for issue to surface or simulate in the near future. Feel free to assign

**Comment by Teeko1993** [comment]
Discord disconnect did not work properly, giving me the error: 

```
2019-04-12 00:07:55,593 - INFO - core.aochat.bot - From Luck: discord disconnect
2019-04-12 00:07:55,716 - ERROR - core.command_service - error processing command: discord disconnect
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\New Orgbots\Pealthunder\Tyrbot\core\command_service.py", line 177, in process_command
    response = handler["callback"](CommandRequest(channel, sender, reply), *self.process_matches(matches, handler["params"]))
  File "C:\Users\Administrator\Desktop\New Orgbots\Pealthunder\Tyrbot\modules\standard\discord\discord_controller.py", line 165, in discord_disconnect_cmd
    self.disconnect_discord_client()
  File "C:\Users\Administrator\Desktop\New Orgbots\Pealthunder\Tyrbot\modules\standard\discord\discord_controller.py", line 347, in disconnect_discord_client
    self.client.loop.create_task(self.client.logout())
  File "C:\Users\Administrator\Desktop\New Orgbots\Pealthunder\Tyrbot\win32\Python3.6\Lib\asyncio\base_events.py", line 302, in create_task
    self._check_closed()
  File "C:\Users\Administrator\Desktop\New Orgbots\Pealthunder\Tyrbot\win32\Python3.6\Lib\asyncio\base_events.py", line 377, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed

2019-04-12 00:07:55,716 - INFO - core.aochat.bot - To Luck: There was an error processing your request.
```

**Comment by Dustify** [comment]
@Teeko1993 was discord in 'broken' state when you did this? Just tested myself and the commands are working okay - bot was connected without issue though

**Comment by Teeko1993** [comment]
Yes it was. The bot thinks it reconnected, but it hasn't. So when trying to manually do !discord disconnect, the bot thinks it is connected, but it can't disconnect because it isn't actually connected. Or atleast that's what it seems to be?

**Comment by Dustify** [comment]
Okay, will see if there's a reliable way of recreating the issue and report back

**Comment by Dustify** [comment]
I've tried blocking comms to discord via firewall, we do get errors but when unblocked the bot recovers gracefully - I will monitor my own instance of the bot and see if the error crops up.

I see disconnections very frequently on another discord bot that I run so I'm not surprised this is an issue - it was painful to resolve in .NET, I ended up wrapping everything in a loop that checked if the client was connected every 30 seconds - if not it would kill and reinitialise everything.

Will report back in the near future :D

**Comment by Dustify** [comment]
Okay I've seen this in my logs but looks like it recovered fine:

![image](https://user-images.githubusercontent.com/626198/56097662-5f440a00-5eef-11e9-98fa-8c7dae40e1af.png)

@Teeko1993 Is the issue recurring for you?

**Comment by Teeko1993** [comment]
That is correct, however if you check the discord connection at that point, the bot thinks it is connected. However it is not.

**Comment by bigwheels16** [comment]
it seems the disconnect issues have been resolved, at least for the time being, so closing this ticket.  will create a new ticket if the issues re-surface.
